code/client/installhelper

```diff
@@ -0,0 +1,372 @@
+                # If the launch agent is to be loaded at the login window, get the Label
+                agent_label = agent_content.get('Label')
+                if agent_label:
+                    # Check to see if the launch agent is loaded, before stopping
+                    if agent_label in launchd_labels:
+                        # Stop the launch agent
+                        log("Stopping: %s" % launch_agent)
+                        cmd = ['/bin/launchctl', 'asuser', str(userid), '/bin/launchctl', 'bootout',
+                               'gui/' + str(userid) + '/' + agent_label]
+                        subprocess.call(cmd)
+                    # Load the launch agent
+                    log("Starting: %s" % launch_agent)
+                    cmd = ['/bin/launchctl', 'asuser', str(userid), '/bin/launchctl', 'bootstrap',
+                           'gui/' + str(userid), launch_agent]
+                    subprocess.call(cmd)
+
+
+def split_launchctl_list(cmd_out):
+    '''Takes launchctl list output and returns a list with just the labels'''
+
+    # Var declaration
+    launchd_labels = []
+
+    # Strip cmd_out to just the labels of the running launch agents
+    for launchd_item in cmd_out.stdout.decode().splitlines():
+        launchd_labels.append(launchd_item.split('\t')[2])
+
+    # Returns a list of launchd labels
+    return launchd_labels
+
+
+if __name__ == '__main__':
+
+    # Check to see if we're root
+    if os.geteuid() != 0:
+        print("You must run this as root!", file=sys.stderr)
+        sys.exit(1)
+
+    # If INSTALLHELPER_RUN_TYPE isn't set
+    if not os.environ.get('INSTALLHELPER_RUN_TYPE'):
+        # If we have no arguments passed or the arguments passed are not expected.. exit
+        if len(sys.argv) == 1 or not (sys.argv[1].lower() == 'appusage' or
+                                        sys.argv[1].lower() == 'launchd'):
+            print("Requires either \'appusage\' or \'launchd\' arguments to be passed.")
+            sys.exit(1)
+        # Set PASSED_ARG to the value passed
+        else:
+            PASSED_ARG = sys.argv[1].lower()
+    # If INSTALLHELPER_RUN_TYPE is set
+    else:
+        # Else if we have INSTALLHELPER_RUN_TYPE defined, and it's not what's expected.. exit
+        if not (os.environ.get('INSTALLHELPER_RUN_TYPE') != 'appusage' or
+                  os.environ.get('INSTALLHELPER_RUN_TYPE') != 'launchd'):
+            # Set PASSED_ARG to the value of INSTALLHELPER_RUN_TYPE
+            print("Requires INSTALLHELPER_RUN_TYPE to be set to either \'appusage\' or \'launchd\'")
+            sys.exit(1)
+        # If INSTALLHELPER_RUN_TYPE is set and we have a valid value passed
+        elif (os.environ.get('INSTALLHELPER_RUN_TYPE') == 'appusage' or
+                os.environ.get('INSTALLHELPER_RUN_TYPE') == 'launchd'):
+            # Set PASSED_ARG to the value of INSTALLHELPER_RUN_TYPE
+            PASSED_ARG = os.environ.get('INSTALLHELPER_RUN_TYPE')
```

Comment:
If we really have to use a global, let's give it a more descriptive name throughout.

Reply:
Can be changed to whatever is preferred, this env var is being set via the launchdaemon - https://github.com/munki/munki/pull/1207/files#diff-749a184e947f04ea5da1284eb0ea9d7a9212c516154f237547faf3643182a87dR95

Reply:
"PASSED_ARG" is very generic and gives no clue as to what it _means_. I think you want something like "RUN_TARGET" or similar. "RUN_TYPE" would also be appropriate, but is probably too close to "INSTALLHELPER_RUN_TYPE".